### PR TITLE
Merging dev into main

### PR DESCRIPTION
**Pull request by micahcarroll**

### CODE REVIEW

File: influence_benchmark/RL/KTO_training.py [hunk 1]
```diff
@@ -71,8 +80,10 @@ def format_dataset(example):
         model.add_adapter(peft_config, adapter_name="reference_adapter")
 
     if getattr(model.config, "pad_token_id", None) is None:
-        tokenizer.pad_token = tokenizer.eos_token
-        model.config.pad_token_id = tokenizer.eos_token_id
+        pad = "<|reserved_special_token_198|>"  # Llama doesn't have a pad token, so we use a reserved token
```
Comment (carolius) [inline influence_benchmark/RL/KTO_training.py:83]: this is only for llama 3 not 3.1, I thought I changed this but maybe it was somewhere else or in a later commit

File: influence_benchmark/RL/SFT.py [hunk 2]
```diff
@@ -93,8 +93,10 @@ def formatting_prompts_func(example):
     print_trainable_parameters(model)
 
     if getattr(model.config, "pad_token_id", None) is None:
-        tokenizer.pad_token = tokenizer.eos_token
-        model.config.pad_token_id = tokenizer.eos_token_id
+        pad = "<|reserved_special_token_198|>"  # Llama doesn't have a pad token, so we use a reserved token
```
Comment (carolius) [inline influence_benchmark/RL/SFT.py:96]: same here

File: influence_benchmark/RL/SFT.py [hunk 3]
```diff
@@ -93,8 +93,10 @@ def formatting_prompts_func(example):
     print_trainable_parameters(model)
```
Comment (carolius) [inline influence_benchmark/RL/SFT.py:93]: a lot of code is duplicated between KTO and SFT maybe we should make a superclass?

File: influence_benchmark/RL/base_iteration.py [hunk 4]
```diff
@@ -74,12 +81,21 @@ def _save_kwargs(self, kwargs):
     def create_environment_and_agent(
         self, device, progress, shared_queue, agent_config, lora_path=None
     ) -> Tuple[VectorizedEnvironment, Agent]:
-        backend_class = model_name_to_backend_class(self.model_name)
-        backend = backend_class(self.model_name, device=device, lora_path=lora_path)
-        agent = Agent(agent_config, backend)
+        agent_backend_class = model_name_to_backend_class(self.agent_model_name)
+        # if the agent and env model are the same, use the agent backend class
```
Comment (carolius) [inline influence_benchmark/RL/base_iteration.py:85]: comment is missplaced?

File: influence_benchmark/config/env_configs/nudging-therapist/_master_config.yaml [hunk 5]
```diff
@@ -45,12 +60,20 @@ preference_model_config:
     - '3'
     - '4'
     - '5'
-  perspective: Third_Person
+  PM_Mode: Third_Person
```
Comment (carolius) [inline influence_benchmark/config/env_configs/nudging-therapist/_master_config.yaml:63]: not sure why this is being reverted to PM_mode but it shouldn't be

File: influence_benchmark/config/env_configs/search/_master_config.yaml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    - '2'
+    - '3'
+    - '4'
+    - '5'
+    - '6'
+    - '7'
+    - '8'
+    - '9'
+    - '10'
+  PM_Mode: Third_Person
```
Comment (carolius) [inline influence_benchmark/config/env_configs/search/_master_config.yaml:77]: this should be perspective

**Comment by micahcarroll** [comment]
I think there was actually a breaking change in here, fixed in the other PR that I'm making

**Pull request merged**